src/Workspaces/VisualBasic/Portable/LanguageServices/VisualBasicSyntaxFactsService.vb

```diff
@@ -1944,5 +1944,25 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
         Public Shadows Function SpansPreprocessorDirective(tokens As IEnumerable(Of SyntaxToken)) As Boolean
             Return MyBase.SpansPreprocessorDirective(tokens)
         End Function
+
+        Public Function IsPartOfPropertyDeclarationHeader(node As SyntaxNode) As Boolean Implements ISyntaxFactsService.IsPartOfPropertyDeclarationHeader
+            Dim propertyDeclaration = node.GetAncestor(Of PropertyStatementSyntax)()
+
+            If propertyDeclaration Is Nothing Then
+                Return False
+            End If
+
+            Dim start = If(propertyDeclaration.AttributeLists.LastOrDefault()?.GetLastToken().GetNextToken().SpanStart, propertyDeclaration.SpanStart)
+            Dim [end] = If(propertyDeclaration.AsClause?.FullSpan.[End], propertyDeclaration.Identifier.FullSpan.End)
```

Comment:
is this the previous code?  did we not handle the case of parameters there?

Reply:
As of right now, not necessarily. Right now the attributes-handling within the service is used only for edge climbing, not filtering locations within nodes, …

It's possible to refactor it so that we wouldn't need to handle attributes here but since the helper is here currently it should work by itself and we don't consider Attributes part of the header. Also it would feel weird asking the service IsOnHeader -> getting yes -> checking if the it's not in Attributes, …

Also, it doesn't handle only attributes, notice the EndBound is at AsClause/FullSpan.End not end of the node (similarly for C#).

The same code is in CSharpService, btw.